TemplePlus/turn_based.cpp

```diff
@@ -120,8 +120,15 @@ void TurnBasedSys::InitiativeListNextActor()
 				dispatch.DispatcherProcessor(dispatcher, dispTypeInitiative, 0, 0);
 			}	
 		}
-		actorInitiative = nextInitiativeIdx = 0;
-		InitiativeRefresh(actorInitiative, 0);
+		nextInitiativeIdx = 0;
```

Comment:
I must confess I don't fully understand what was going on here, and in general these sorts of changes have to be done very carefully.
Also, I don't know if it's an issue, but have you considered/checked what happens with concurrent turns enabled?

Reply:
This line was already changed from the original code. Can you explain why?

In the original code, it sets `nextInitiativeIdx` to 0, then calls `InitiativeRefresh`, then it sets `actorInitiative` to 0. The effect this has is sending a signal that counts from the current initiative to 0, so that after this when you count from 0 to the max initiative, no numbers have been skipped.  In TemplePlus, instead there is a signal sent with two 0s, which was actually the reason that Stunned was broken.

However, an actual initiative can be 0 or negative, which is why I added the check after these lines, because they will cause signals that count from, say, -1 to 0, before wrapping around, which have the same bad effects as the 0->0 signal, but are harder to trigger.

Reply:
Oh, I should also mention, one of the reasons I fixed this is that there could otherwise be situations where someone is stunned permanently, because the stunned condition is set to tick during an initiative number that is skipped. You could do this by stunning someone as the lowest initiative, then delaying your turn so that you have a higher initiative in the next round. Then your previous initiative is skipped for the rest of combat, and the stun would not tick anymore.

Reply:
Huh, looks like a simple mistake. I only replaced that function for debugging apparently (the irony :P)
Looking at the condition list, Stunned is the only one that reacts to S_Initiative_Update.